packages/modal/src/ModalContainer.spec.tsx

```diff
@@ -52,6 +52,14 @@ describe('FocusJailContainer', () => {
       userEvent.click(getByTestId('backdrop'));
       expect(onCloseSpy).toHaveBeenCalled();
     });
+
+    it('does not call onClose when inital click occurs within modal', () => {
+      const { getByTestId } = render(<BasicExample onClose={onCloseSpy} />);
+
+      fireEvent.mouseDown(getByTestId('modal'));
+      fireEvent.mouseUp(getByTestId('backdrop'));
+      expect(onCloseSpy).not.toHaveBeenCalled();
+    });
```

Comment:
When the changeset in `useModal.ts` is reverted, this test still passes. Would that mean that this test would not have caught the bug?

Reply:
The original events used `onClick`, which wouldn't be triggered with this exact test. I had to split this into two with `fireEvent` to trigger the correct flows.

If you were to remove the following check this would fail the test:
```
if (!isModalMousedDownRef.current) {
```

Reply:
Ah, gotcha thanks. LGTM!